linalg/sparsemat.cpp

```diff
@@ -121,16 +121,17 @@ SparseMatrix::SparseMatrix(int nrows, int ncols, int rowsize)
    }
 }
 
-SparseMatrix::SparseMatrix(const SparseMatrix &mat, bool copy_graph)
+SparseMatrix::SparseMatrix(const SparseMatrix &mat, bool copy_graph,
+                           MemoryType mt)
    : AbstractSparseMatrix(mat.Height(), mat.Width())
 {
    if (mat.Finalized())
    {
       const int nnz = mat.I[height];
       if (copy_graph)
       {
-         I.New(height+1, mat.I.GetMemoryType());
-         J.New(nnz, mat.J.GetMemoryType());
+         I.New(height+1, mt == MemoryType::SIZE ? mat.I.GetMemoryType() : mt);
+         J.New(nnz, mt == MemoryType::SIZE ? mat.J.GetMemoryType() : mt);
```

Comment:
If I understand correctly you're changing the semantic of `MemoryType::SIZE` to mean "use the same `MemoryType` as the given SparseMatrix", this should be documented.

Reply:
Yes, good call, I'll add some documentation